### PR TITLE
Bugfix/#10441 correctly validate second colon in jdbc url parameter in Java client.

### PR DESCRIPTION
**Pull request by peteraisher**
Add a test case and a bugfix for the incomplete validation of the JDBC `url` parameter in `org.duckdb.DuckDBConnection#newConnection`.

### CODE REVIEW

File: tools/jdbc/src/test/java/org/duckdb/TestDuckDBJDBC.java [hunk 1]
```diff
@@ -2150,6 +2150,14 @@ public static void test_connect_wrong_url_bug848() throws Exception {
         assertNull(d.connect("jdbc:h2:", null));
     }
 
+    public static void test_new_connection_wrong_url_bug10441() throws Exception {
+        try {
+            DuckDBConnection.newConnection("jdbc:duckdb@", false, new Properties());
```
Comment (Mause) [inline tools/jdbc/src/test/java/org/duckdb/TestDuckDBJDBC.java:2155]: Can you use the assertThrows method here instead?

**Review by Mause — CHANGES_REQUESTED** [review]

File: tools/jdbc/src/test/java/org/duckdb/TestDuckDBJDBC.java (hunk 1 above)
Comment (peteraisher) [reply tools/jdbc/src/test/java/org/duckdb/TestDuckDBJDBC.java:2155]: With pleasure!

I saw the `try { …; fail(); } catch (…) { }` pattern used in some of the first few test methods in the class — is that something to tidy up in a separate PR?

Comment (Mause) [reply tools/jdbc/src/test/java/org/duckdb/TestDuckDBJDBC.java:2155]: Yeah we need to go back and remove those old usages

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**